pennylane_lightning/core/_serialize.py

```diff
@@ -254,23 +260,31 @@ def _pauli_sentence(self, observable, wires_map: dict = None):
         terms = [self._pauli_word(pw, wires_map) for pw in pwords]
         coeffs = np.array(coeffs).astype(self.rtype)
 
+        # TODO: Add this
+        # if len(terms) == 1 and coeffs[0] == 1.0:
+        #     return terms[0]
+
         if self.split_obs:
             return [self.hamiltonian_obs([c], [t]) for (c, t) in zip(coeffs, terms)]
         return self.hamiltonian_obs(coeffs, terms)
 
-    # pylint: disable=protected-access
+    # pylint: disable=protected-access, too-many-return-statements
     def _ob(self, observable, wires_map: dict = None):
         """Serialize a :class:`pennylane.operation.Observable` into an Observable."""
-        if isinstance(observable, Tensor):
+        if isinstance(observable, (Prod, Sum, SProd)) and observable.pauli_rep is not None:
+            return self._pauli_sentence(observable.pauli_rep, wires_map)
```

Comment:
This seems to be redundent with another if-condition (line 284). this can be re-written in some format closer to Catalyst as we discussed.

Reply:
The reason for this was because to minimize the number of tests that needed to be updated, only the new op math observables (except for `LinearCombination`) needed to use the pauli rep. Another thing that I'd like to leave as it for now and address in the next PR. The plan is to use a similar structure to Catalyst in that PR anyway, so this will become a non-issue.